k8s-model-server/README.md

```diff
@@ -0,0 +1,170 @@
+will be used in the Kubernetes Deployment's PodSecurityContext so that the container is not run with root privileges
+(This is considered to be a best practice).
+
+### Build the Image
+A [Makefile](docker/Makefile) is included that will build the image and push it to private hosting inside your project. 
+
+Download and install the [Google Cloud SDK](https://cloud.google.com/sdk/downloads) (Software Development Kit).
+
+Enable the [GCR](https://cloud.google.com/container-registry/docs/quickstart) (Google Container Registry) API in 
+your project.
+
+Set the name of your project at the top of the Makefile ```PROJECT_ID=<your-project>```.
+
+The command below will build the docker image.
+
+```commandline
+make all
+```
+
+The command below will push the docker image to your project's GCR.
+
+```commandline
+make push
+```
+
+You can learn more about [building a TensorFlow model server](https://www.tensorflow.org/serving/serving_advanced) and 
+running a [TensorFlow model server in docker](https://www.tensorflow.org/serving/docker) in the TensorFlow Serving 
+documentation.
+
+## Create a Bucket
+You are going to serve the model from [GCS](https://cloud.google.com/storage/). 
+
+Download and install the [Google Cloud SDK](https://cloud.google.com/sdk/downloads) (Software Development Kit).
+
+Use the [gsutil mb](https://cloud.google.com/storage/docs/gsutil/commands/mb) command to create a bucket. Note that 
+the bucket name must be globally unique.
+```commandline
+gsutil mb gs://<ucket-name>
+``` 
+
+
+## Upload a Model
+You can train an inception model using the instructions from the 
+[TensorFlow documentation](https://www.tensorflow.org/tutorials/image_retraining), or you can download the pre-trained
+[example](inception.tar.gz).
+
+If you downloaded the example decompress it.
+```commandline
+tar -zxpf inception.tar.gz
+```
+
+Use [gsutil cp](https://cloud.google.com/storage/docs/gsutil/commands/cp) to upload the inception model to the cloud 
+storage bucket you created above.
+```commandline
+gsutil cp -r inception gs://<bucket-name>
+```
+
+Use [gsutil_ls](https://cloud.google.com/storage/docs/gsutil/commands/ls) to view the contents of your bucket. You 
+will see that the contents of the model are stored in the `gs://<bucket-name>/inception/1` directory. This is the 
+first version of the model that we will serve.
+
+```commandline
+gsutil ls -r gs://<bucket-name>/ 
+gs://<bucket-name>/inception/:
+gs://<bucket-name>/inception/
+
+gs://<bucket-name>/inception/1/:
+gs://<bucket-name>/inception/1/
+gs://<bucket-name>/inception/1/saved_model.pb
+
+gs://<bucket-name>/inception/1/variables/:
+gs://<bucket-name>/inception/1/variables/variables.data-00000-of-00001
+gs://<bucket-name>/inception/1/variables/variables.index
+```
+
+## Create a Kubernetes Cluster
+
+Use [gcloud container clusters create](https://cloud.google.com/sdk/gcloud/reference/container/clusters/create) to turn 
```

Comment:
Why not link to GKE documentation? I'd also include a GPU guide here. Serving TF using CPUs is probably better suited for Lambda or App engine as long as the model fits within the resource constraints. I feel GPU or on-prem is where k8s shines.